msticpy/nbtools/timeline.py

```diff
@@ -0,0 +1,940 @@
+    tool_tip_columns: Set[str] = set()
+    min_time = pd.Timestamp(pd.Timestamp.max)
+    max_time = pd.Timestamp(pd.Timestamp.min)
+    y_index = 0
+    # Shift the Viridis palatte so we lose the top, harder-to-see colors
+    series_count = len(data)
+    palette_size = min(256, series_count + int(series_count / 2))
+    colors = viridis(palette_size)
+    for ser_name, series_def in data.items():
+        data_columns: Set[str] = set()
+        series_data = series_def["data"]
+
+        # if the series has source columns, use those
+        src_cols = series_def.get("source_columns", def_source_columns)
+        data_columns.update(src_cols if src_cols else def_source_columns)
+
+        # add these columns to the tool tip column set
+        tool_tip_columns.update(data_columns)
+
+        time_col = series_def.get("time_column", None)
+        if not time_col:
+            time_col = time_column
+            series_def["time_column"] = time_col
+
+        min_time = min(min_time, series_data[time_col].min())
+        max_time = max(max_time, series_data[time_col].max())
+        data_columns.update([time_col])
+        # Create the Column data source to plot
+        graph_df = series_data[list(data_columns)].copy()
+        graph_df["y_index"] = y_index
+        series_def["y_index"] = y_index
+        ser_color = series_def.get("color", None)
+        if not ser_color:
+            ser_color = colors[y_index % palette_size]
+            series_def["color"] = ser_color
+        # Wrap tooltip lines longer than 50 chars
+        _wrap_df_columns(graph_df, 50)
+        series_def["source"] = ColumnDataSource(graph_df)
+        y_index += 1
+
+    # build the tool tips from all specified columns
+    tool_tip_items = [(f"{col}", f"@{col}") for col in tool_tip_columns]
+    hover = HoverTool(tooltips=tool_tip_items, formatters={"Tooltip": "printf"})
+
+    title = f"Timeline: {title}" if title else "Event Timeline"
+    start_range = min_time - ((max_time - min_time) * 0.1)
+    end_range = max_time + ((max_time - min_time) * 0.1)
+    height = height if height else _calc_auto_plot_height(len(data))
+    plot = figure(
+        x_range=(start_range, end_range),
+        min_border_left=50,
+        plot_height=height,
+        plot_width=width,
+        x_axis_label="Event Time",
+        x_axis_type="datetime",
+        x_minor_ticks=10,
+        tools=[hover, "xwheel_zoom", "box_zoom", "reset", "save", "xpan"],
+        title=title,
+    )
+    if show_yaxis:
+        plot.yaxis.visible = True
+        if data:
+            y_labels = {ser_def["y_index"]: str(lbl) for lbl, ser_def in data.items()}
+            plot.yaxis.major_label_overrides = y_labels
+    plot.ygrid.minor_grid_line_color = "navy"
+    plot.ygrid.minor_grid_line_alpha = 0.3
+    plot.xgrid.minor_grid_line_color = "navy"
+    plot.xgrid.minor_grid_line_alpha = 0.3
+    # Create plot bar to act as as range selector
+    rng_select = _create_range_tool(
```

Comment:
We make this optional for the other graphs, we should probably do the same here.